### PR TITLE
Music now crossfades to victory/defeat tune

### PR DESCRIPTION
**Pull request by dspages**
Also fixed a bug where a player-controlled unit's attack type wasn't getting reset when the attack was used, resulting in that same attack being used for future attacks.

### CODE REVIEW

File: Assets/Scripts/Combat/MusicManager.cs [hunk 1]
```diff
@@ -14,34 +15,49 @@ public class MusicManager : MonoBehaviour
...
+        while (elapsed < duration)
+        {
+            aud[0].volume = 1f - (elapsed / duration);
+            aud[1].volume = elapsed / duration;
+            elapsed += Time.deltaTime;
+            yield return null;
+        }
+        aud[1].volume = 1f;
+        aud[0].volume = 0f;
+        yield break;
```
Comment (basile-laderchi) [inline Assets/Scripts/Combat/MusicManager.cs:52]: Before exiting the crossFade method doesn't the aud[0] need to be stopped?

File: Assets/Scripts/Combat/MusicManager.cs [hunk 2]
```diff
@@ -14,34 +15,49 @@ public class MusicManager : MonoBehaviour
...
     {
         if (gameOver) return;
-        AudioSource thisSource = GetComponent<AudioSource>();
-        thisSource.clip = DefeatSong;
-        thisSource.Play();
+        StartCoroutine(crossFade(1.0f, DefeatSong));
         gameOver = true;
     }
 
+    IEnumerator crossFade(float duration, AudioClip fadeTo)
```
Comment (basile-laderchi) [inline Assets/Scripts/Combat/MusicManager.cs]: Rename the method to CrossFade (first letter capitalization).

**Review by basile-laderchi — CHANGES_REQUESTED** [review]

File: Assets/Scripts/Combat/MusicManager.cs (hunk 1 above)
Comment (dspages) [reply Assets/Scripts/Combat/MusicManager.cs:52]: Yeah, that's a better way to do it for sure.

File: Assets/Scripts/Combat/MusicManager.cs (hunk 2 above)
Comment (dspages) [reply Assets/Scripts/Combat/MusicManager.cs]: Good call, thanks!

**Pull request merged**